pkg/codegen/docs/gen.go

```diff
@@ -486,6 +486,40 @@ func (mod *modContext) withDocGenContext(dctx *docGenContext) *modContext {
 	return &newctx
 }
 
+func (dctx *docGenContext) getSupportedLanguages(isOverlay bool, overlaySupportedLanguages []string) []string {
+	// By default, all languages are supported.
+	if !isOverlay || len(overlaySupportedLanguages) == 0 {
+		return dctx.supportedLanguages
+	}
+	var supportedLanguages []string
+	allLanguages := codegen.NewStringSet(dctx.supportedLanguages...)
+	// Ensure that the overlay languages are a subset of the supported languages.
+	for _, lang := range overlaySupportedLanguages {
+		if allLanguages.Has(lang) {
+			supportedLanguages = append(supportedLanguages, lang)
+		}
+	}
+
+	return supportedLanguages
+}
```

Comment:
So for this, I think we want to ensure that the order we spit them out in is the same as it is today. I think this could reduce the diffs in the tests, but also I think it'll be the behaviour we want, since TypeScript and Python "should" come first (see all our docs, etc.) -- they are our most popular languages.

Reply:
The order in the docs stay the same. That's encoded in the Langchooser component of the registry

Reply:
Ah pleasing -- my bad

Reply:
Here's where that's happening in the registry: https://github.com/pulumi/registry/blob/0fe89140b25a33f57f1d1fe10ddb96a0d1d49aca/themes/default/theme/stencil/src/components/chooser/chooser.tsx#L290